nixos/modules/security/apparmor/profiles.nix

```diff
@@ -0,0 +1,335 @@
+    ${etcRule "services"}
+
+    ${etcRule "default/nss"}
+
+    # libnl-3-200 via libnss-gw-name
+    ${etcRule {path="libnl"; trail="/classid";}}
+    ${etcRule {path="libnl-3"; trail="/classid";}}
+
+    mr ${getLib pkgs.nss}/lib/libnss_*.so*,
+    mr ${getLib pkgs.nss}/lib64/libnss_*.so*,
+  '';
+  "abstractions/nis" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/nis
+  '';
+  "abstractions/nvidia" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/nvidia
+    ${etcRule "vdpau_wrapper.cfg"}
+  '';
+  "abstractions/opencl-common" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/opencl-common
+    ${etcRule {path="OpenCL"; trail="/**";}}
+  '';
+  "abstractions/opencl-mesa" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/opencl-mesa
+    ${etcRule "default/drirc"}
+  '';
+  "abstractions/openssl" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/openssl
+    ${etcRule {path="ssl"; trail="/openssl.cnf";}}
+  '';
+  "abstractions/p11-kit" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/p11-kit
+    ${etcRule {path="pkcs11"; trail="/";}}
+    ${etcRule {path="pkcs11"; trail="/pkcs11.conf";}}
+    ${etcRule {path="pkcs11"; trail="/modules/";}}
+    ${etcRule {path="pkcs11"; trail="/modules/*";}}
+  '';
+  "abstractions/perl" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/perl
+    ${etcRule {path="perl"; trail="/**";}}
+  '';
+  "abstractions/php" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/php
+    ${etcRule {path="php"; trail="/**/";}}
+    ${etcRule {path="php5"; trail="/**/";}}
+    ${etcRule {path="php7"; trail="/**/";}}
+    ${etcRule {path="php"; trail="/**.ini";}}
+    ${etcRule {path="php5"; trail="/**.ini";}}
+    ${etcRule {path="php7"; trail="/**.ini";}}
+  '';
+  "abstractions/postfix-common" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/postfix-common
+    ${etcRule "mailname"}
+    ${etcRule {path="postfix"; trail="/*.cf";}}
+    ${etcRule "postfix/main.cf"}
+    ${etcRule "postfix/master.cf"}
+  '';
+  "abstractions/python" = ''
+    #include ${pkgs.apparmor-profiles}/etc/apparmor.d/abstractions/python
+    ${etcRule {path="python2.4"; trail="/**";}}
+    ${etcRule {path="python2.5"; trail="/**";}}
+    ${etcRule {path="python2.6"; trail="/**";}}
+    ${etcRule {path="python2.7"; trail="/**";}}
+    ${etcRule {path="python3.0"; trail="/**";}}
+    ${etcRule {path="python3.1"; trail="/**";}}
+    ${etcRule {path="python3.2"; trail="/**";}}
+    ${etcRule {path="python3.3"; trail="/**";}}
+    ${etcRule {path="python3.4"; trail="/**";}}
+    ${etcRule {path="python3.5"; trail="/**";}}
+    ${etcRule {path="python3.6"; trail="/**";}}
+    ${etcRule {path="python3.7"; trail="/**";}}
+    ${etcRule {path="python3.8"; trail="/**";}}
+    ${etcRule {path="python3.9"; trail="/**";}}
```

Comment:
I suppose this will need to be updated whenever there's a new version of Python?  If AppArmor can't be made to allow all "pythonX.Y", should there be a comment saying to update this list added somewhere where someone adding a new Python version to nixpkgs will see it?

Reply:
I would say that this profile falls under the [introductive comment](https://github.com/NixOS/nixpkgs/pull/93457/commits/1c09929bea22619bdd522afb44600f7b775dc9a2#diff-0a843295647e2675849451220fba266dR16) saying that I generally don't know the rules which are relevent, and likely that those current rules are incorrect or useless. Hence I would not add a comment to say what to do except investigate on how `/etc/pythonX.Y` is generated on NixOS, if it ever is. Those present rules match a configuration where `/etc/pythonX.Y` is a symlink, like it is for instance  the case with `/etc/systemd/system -> /etc/static/systemd/system`, but maybe NixOS/Python people would generate it otherwise, eg. `environment.etc."pythonX.Y/foo".text = "bar"`. Maybe we should simply remove those rules and let correct ones be added by a later PR once the `/etc/pythonX.Y` setup has been elucidated.

Reply:
I grepped `pkgs/development/interpreters/python` and `pkgs/development/python-modules`, and I see no mention of any `/etc/python*` directory, and I see that I don't have any of these directories.

@FRidh, as Python maintainer, can you confirm that NixOS doesn't use these `/etc/pythonX.Y` directories?

Reply:
No we don't use `/etc/pythonX.Y`. What kind of system-wide Python configuration could one expect?

Reply:
I don't know why this is needed, but at least remove all old Python versions.

Reply:
Thanks for confirming.  I guess these rules aren't needed then, though I'm not very familiar with either Python or AppArmor.